ShadowDOM/explainer.md

```diff
@@ -0,0 +1,395 @@
+* "Shadow DOM encapsulation is too much. E.g. No way to adopt form styling from the surrounding page for common elements (buttons, inputs, etc) unless I'm willing to put them in light DOM"
+
+For additional use cases, please see issue [939](https://github.com/WICG/webcomponents/issues/939) 
+
+## Alternatives to using style in DSD 
+### Constructable Stylesheets 
+Developers can create stylesheets that can be applied to multiple shadow roots, using existing JavaScript, as outlined by the example below. 
+
+Step 1: Create a new Constructable Stylesheet: 
+```js
+const constStylesheet = new CSSStyleSheet(); 
+```
+Step 2: Add styles to the Constructable Stylesheet: 
+```js
+constStylesheet.replaceSync(` 
+  .my-button { 
+    background-color: #0074D9; 
+  } 
+`); 
+```
+Step 3: Attach the Constructable Stylesheet to the shadow root: 
+```js
+shadow.adoptedStyleSheets = [constStylesheet]; 
+```
+The downside of this approach is a potential FOUC, where the element is initially painted without styles, and then repainted with the Constructable Stylesheet. 
+
+### Using `rel="stylesheet"` attribute
+Using `<link rel="stylesheet">` to share styles across Shadow DOM boundaries helps maintain consistent design, reducing style duplication and potentially shrinking component sizes for faster load times. However, it can cause redundant network requests since each component that uses `<link rel="stylesheet">` within its Shadow DOM may trigger an expensive operation such as a network request or a disk access.   
+
+### CSS `@import` rules
+Global styles can be included in a single stylesheet, which is then importable into each shadow root to avoid redundancy. Inline `<style>` blocks do not support `@import` rules, so this approach must be combined with either of the aforementioned Constructable Stylesheets or `<link rel>` approaches. If the stylesheet is not already loaded, this could lead to an FOUC.  
+
+## Proposal: Inline, declarative CSS module scripts   
+This proposal builds on [CSS module scripts](https://web.dev/articles/css-module-scripts), enabling authors to declare a CSS module inline in an HTML file and link it to a DSD using its [module specifier](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules#:~:text=The-,module%20specifier,-provides%20a%20string). A `type=”css-module”` attribute on the `<script>` element would define it as a CSS  module script and the specifier attribute would add it to the module cache as if it had been imported. This allows the page to render with the necessary CSS modules attached to the correct scopes without needing to load them multiple times. Note that module maps are global, meaning that modules defined in a Shadow DOM will be accessible throughout the document context. 
```

Comment:
Does `type="css-module"` mean that the rules in the `<style>` element apply *only* to shadow DOMs that adopt the sheet and *not* to the main document? Either way, can you clarify this aspect of the behavior in the explainer text?